Can you look at the failure, please?

`//tensorflow/python/lib/io:file_io_test` is failing and the failure looks related to the PR

```
[       OK ] FileIoTest.test_session
======================================================================
ERROR: testFileSeekableWithZip (__main__.FileIoTest)
testFileSeekableWithZip (__main__.FileIoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/lib/io/file_io_test.runfiles/org_tensorflow/tensorflow/python/lib/io/file_io_test.py", line 674, in testFileSeekableWithZip
    _ = [i for i in info.items()]
  File "/usr/local/lib/python3.6/dist-packages/numpy/lib/npyio.py", line 249, in items
    return [(f, self[f]) for f in self.files]
  File "/usr/local/lib/python3.6/dist-packages/numpy/lib/npyio.py", line 249, in <listcomp>
    return [(f, self[f]) for f in self.files]
  File "/usr/local/lib/python3.6/dist-packages/numpy/lib/npyio.py", line 228, in __getitem__
    bytes = self.zip.open(key)
  File "/usr/lib/python3.6/zipfile.py", line 1419, in open
    % (zinfo.orig_filename, fname))
zipfile.BadZipFile: File name in directory 'arr_0.npy' and header b'\xbcx\xf3`\xb0\x93\x82P\x0f' differ.

======================================================================
FAIL: testSeek (__main__.FileIoTest)
testSeek (__main__.FileIoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/lib/io/file_io_test.runfiles/org_tensorflow/tensorflow/python/lib/io/file_io_test.py", line 462, in testSeek
    self.assertEqual(18, f.tell())
AssertionError: 18 != 27

======================================================================
FAIL: testSeekFromWhat (__main__.FileIoTest)
testSeekFromWhat (__main__.FileIoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/lib/io/file_io_test.runfiles/org_tensorflow/tensorflow/python/lib/io/file_io_test.py", line 496, in testSeekFromWhat
    self.assertEqual(18, f.tell())
AssertionError: 18 != 27

----------------------------------------------------------------------
Ran 65 tests in 0.947s

FAILED (failures=2, errors=1)
================================================================================
```

--- a/tensorflow/core/lib/io/buffered_inputstream.cc
+++ b/tensorflow/core/lib/io/buffered_inputstream.cc
@@ -157,15 +157,22 @@ Status BufferedInputStream::Seek(int64 position) {
                                    position);
   }
 
-  // Position of the buffer within file.
-  const int64 bufpos = Tell();
-  if (position < bufpos) {
-    // Reset input stream and skip 'position' bytes.
+  // Position of the buffer's lower limit within file.
+  const int64 buf_lower_limit  = input_stream_->Tell() - limit_ ;
+  if (position < buf_lower_limit) {
+    // Seek before buffer, reset input stream and skip 'position' bytes.
     TF_RETURN_IF_ERROR(Reset());
     return SkipNBytes(position);
   }
 
-  return SkipNBytes(position - bufpos);
+  if (position < Tell()) {
+    // Seek within buffer before 'pos_'
+    pos_ -= Tell() - position;
+    return Status::OK();
+  }
+
+  // Seek after 'pos_'
+  return SkipNBytes(position - Tell());
 }
 
 template <typename T>

--- a/tensorflow/core/lib/io/buffered_inputstream_test.cc
+++ b/tensorflow/core/lib/io/buffered_inputstream_test.cc
@@ -394,6 +394,29 @@ TEST(BufferedInputStream, Seek) {
   }
 }
 
+TEST(BufferedInputStream, Seek_NotReset) {
+  // This test verifies seek backwards within the buffer doesn't reset input_stream
+  Env* env = Env::Default();
+  string fname;
+  ASSERT_TRUE(env->LocalTempFilename(&fname));
+  TF_ASSERT_OK(WriteStringToFile(env, fname, "0123456789"));
+  std::unique_ptr<RandomAccessFile> file;
+  TF_ASSERT_OK(env->NewRandomAccessFile(fname, &file));
+
+  std::unique_ptr<RandomAccessInputStream> input_stream(
+      new RandomAccessInputStream(file.get()));
+  tstring read;
+  BufferedInputStream in(input_stream.get(), 3);
+
+  TF_ASSERT_OK(in.ReadNBytes(4, &read));
+  int before_tell =  input_stream.get()->Tell();
+  EXPECT_EQ(before_tell, 6);
+  // Seek backwards
+  TF_ASSERT_OK(in.Seek(3));
+  int after_tell = input_stream.get()->Tell();
+  EXPECT_EQ(before_tell, after_tell);
+}
+
 TEST(BufferedInputStream, ReadAll_Empty) {
   Env* env = Env::Default();
   string fname;

